packages/state-manager/src/sagas/identity/identity.selectors.ts

```diff
@@ -37,6 +38,27 @@ export const csr = createSelector(communitiesSelectors.currentCommunityId, selec
   return identities[id]?.userCsr
 })
 
+export const usernameTaken = createSelector(currentIdentity, csrsMapping, (identity, csrs) => {
+  const userCertificate = identity?.userCertificate
+  if (userCertificate) return false
+
+  const username = identity?.nickname
+  if (!username) return false
+
+  const allUsernames: string[] = Object.values(csrs).map(u => u.username)
```

Comment:
These are just CSRs? Shouldn't we be checking against signed certificates? Since a username isn't valid until the certificate has been signed by the owner.

Reply:
Thanks for catching this